### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by Rav23** [highsolutions/eloquent-sequence#9]
Firstly, this package looks very indispensable for a project Im working on.

How would one go about tackling the following:
- generate a sequence for a given model on a monthly basis. IE: while in the current month, it may increment the value. When a record is created in a new month, then the sequence is restarted.

JANUARY:
1
2
3
4

FEBRUARY:
1
2
3
4

Is it possible?

I´ve tried by checking the current month, then checking the last invoice nember, then finally increment / reset the sequence. It works! But after careful analysis I started to notice that at times I get duplicate sequence numbers for the same month.

So Im looking for a mature solution.

**Comment by highsolutions** [comment]
Hello @Rav23 ,
yes - this is possible. We are using our package in one project to manage sequence identifiers for every month separately.

To make what you need, we recommend creating another column in a database where you will store month (yyyy-mm or mm - depends on what you need). And then you point this model attribute as group of sequencing. Simple as that.

If you any more questions, please ask.

**Comment by Rav23** [comment]
Firstly, thank you for the prompt reply.

I will try to implement your suggestion using this package. If I'm successful then I'll post feedback and possibly have this package as a reference to future projects. 

Thank YOU.

**Comment by highsolutions** [comment]
Good luck!

If you have any trouble with our package, post your code. We'll try to help :).

**Comment by Rav23** [comment]
Hi there guys,

Just to post feedback on my earlier request. I followed your suggestion in creating an extra column to house the month format and it did the trick.

My colleague asked to change the sequence so it reads (2, 4, 6, 8) and all I did was hook into the Model::created event and increment the seq by 1.

Sample Model:

`public static function boot()
    {
        parent::boot();

            static::created(function($sample) {

                $sample->seq += 1; 
                $sample->description = $sample->sample_month . '/' . str_pad ($sample->seq, 4, '0', STR_PAD_LEFT);
                $sample->save();
        
            });

    }`

<img width="683" alt="sample_table" src="https://user-images.githubusercontent.com/2884479/41457212-d5bde142-707a-11e8-9792-43c2cf3d1b79.png">

So once again, a HUGE thank you for this awesome package.
